samples/peripheral/802154_phy_test/src/ctrl/ptt_parser.c

```diff
@@ -0,0 +1,232 @@
+ * All rights reserved.
+ *
+ * Use in source and binary forms, redistribution in binary form only, with
+ * or without modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/* Purpose: converting string of ASCII codes to integers */
+
+#include "ptt_parser_internal.h"
+
+#include <stddef.h>
+#include <string.h>
+#include <limits.h>
+#include <stdlib.h>
+
+#define PARSER_DECIMAL_BASE     10
+#define PARSER_HEXADECIMAL_BASE 16
+#define PARSER_BASE_NOT_FOUND   0xffu
+
+static enum ptt_ret_t ptt_parser_hex_string_next_byte(const char *p_hex_str, uint8_t *p_out_num);
+
+enum ptt_ret_t ptt_parser_string_to_int(const char *p_str,
+				   int32_t    *p_out_value,
+				   uint8_t      base,
+				   int32_t      min,
+				   int32_t      max)
+{
+	if (p_str == NULL) {
+		return PTT_RET_NULL_PTR;
+	}
+
+	if (p_out_value == NULL) {
+		return PTT_RET_NULL_PTR;
+	}
+
+	enum ptt_ret_t ret = PTT_RET_SUCCESS;
+
+	char *p_end;
+
+	long out_num = strtol(p_str, &p_end, base);
+
+	/* "In particular, if *nptr is not '\0' but **endptr is '\0' on return, the entire string
+	 * is valid." <- from man strtol
+	 */
+	/* also lets discard LONG_MIN/LONG_MAX values which show strtol failed */
+	if (('\0' != *p_end) ||
+	(out_num < min) || (out_num > max) ||
```

Comment:
Please indent (ideally to the opening bracket). There are mutliple other places like this across the code, please fix them as well.

Reply:
Fixed here and in other occurrences